test/tests/tsdx-build.test.js

```diff
@@ -33,6 +33,44 @@ describe('tsdx build', () => {
     expect(output.code).toBe(0);
   });
 
+  it('should compile multiple entries into a dist directory', () => {
+    util.setupStageWithFixture(stageName, 'build-default');
+
+    const output = shell.exec(
+      [
+        'node ../dist/index.js build',
+        '--entry src/index.ts',
+        '--entry src/foo.ts',
+        '--entry src/subdir1/subdir1-2/index.ts',
+        '--entry src/**/*.ts',
+        '--format esm,cjs',
+      ].join(' ')
+    );
+    const outputFiles = shell.ls('-R', 'dist/');
+
+    function arrToDict(arr) {
+      return arr.reduce((dict, elem) => {
+        dict[elem] = true;
+        return dict;
+      }, {});
+    }
+    const outputDict = arrToDict(outputFiles);
+
+    const entries = ['index', 'foo', 'subdir1/subdir1-2/index', 'subdir1/glob'];
+    const expected = entries.reduce((dict, entry) => {
```

Comment:
IMHO the expected result should not be generated, tests are also documentation and this is not immediately readable, because one needs to evaluate this in their head.

Reply:
Mmmm, I understand where you're coming from, but if each file were manually written out, that would be even less readable, your eyes would just glaze over due to its sheer length and repetitiveness.

Per my commits I do think `toMatchObject` also provides far superior readability during test errors (and will show everything that's missing, not just one file at a time). Was planning on refactoring the other tests to use it once this were merged.

Regarding generation, these are only partially generated; it's just the extensions that are added. I could push these all to an array and then use `arrToDict` for easier readability however.
I've seen tests generate a lot more and snapshots are also s form of generating tests. I don't think they're bad, but should be used with caution

Reply:
> Mmmm, I understand where you're coming from, but if each file were manually written out, that would be even less readable: your eyes would just glaze over due to its sheer length and repetitiveness.

I wasn't referring to the files itself (although their content could be snapshotted) but rather just to the list of output files. I'm fine with `outputDict` being generated, but `expected` could be a plan object.

Reply:
I meant the expected file list too. That plain object would have 25 entries and would be likely to grow. Something that long and repetitive isn't easy on the eyes.